### ISSUE TITLE
Gradle 6.0 compatibility

### ISSUE THREAD

**Issue by msink** [michel-kraemer/gradle-download-task#143]
Tried upgrading Gradle to version 6.0
`./gradlew wrapper --gradle-version=6.0

It works, but with warnings, and without showing download progress:
```
Property 'authScheme' is not annotated with an input or output annotation. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0.
Property 'cachedETagsFile' is not annotated with an input or output annotation. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0.
Property 'connectTimeout' is not annotated with an input or output annotation. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0.
Property 'dest' is not annotated with an input or output annotation. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0.
Property 'downloadTaskDir' is not annotated with an input or output annotation. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0.
Property 'headers' is not annotated with an input or output annotation. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0.
Property 'password' is not annotated with an input or output annotation. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0.
Property 'readTimeout' is not annotated with an input or output annotation. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0.
Property 'retries' is not annotated with an input or output annotation. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0.
Property 'src' is not annotated with an input or output annotation. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0.
Property 'useETag' is not annotated with an input or output annotation. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0.
Property 'username' is not annotated with an input or output annotation. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0.
Property 'acceptAnyCertificate' is not annotated with an input or output annotation. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0.
Property 'compress' is not annotated with an input or output annotation. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0.
Property 'onlyIfModified' is not annotated with an input or output annotation. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0.
Property 'onlyIfNewer' is not annotated with an input or output annotation. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0.
Property 'overwrite' is not annotated with an input or output annotation. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0.
Property 'quiet' is not annotated with an input or output annotation. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0.
Property 'tempAndMove' is not annotated with an input or output annotation. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0.
Unable to get progress logger. Download progress will not be displayed.
```

**Comment by michel-kraemer** [comment]
Thanks for reporting this. I'll look into it as soon as I have time.

**Comment by gradlenoob** [comment]
Hi 

I was trying the same and looked a bit into it. The reason for the not shown progress is that the method setLoggingHeader is not there any more at the target object for invocation in the ProgressLoggerWrapper.

The missing annotations are unclear to me - I tried to add them in DownloadSpec interface but with no success.

**Comment by michel-kraemer** [comment]
I just released version 4.0.2, which addresses both issues. Please test again. Thanks for the help!!
